### PR TITLE
Setup QHacks Bot project

### PR DESCRIPTION
**Pull request by RobertWSaunders**
## Proposed Change

Fixes: #2

This sets up the QHacks Bot project!

### How did you do this?

Copy configuration from other QHacks projects for consistency.

### Why are you choosing this approach?

Start the project!

### Any open questions you want to ask code reviewers?

Nope! 😃 

### List of changes:

  - Creates GitHub templates
  - Adds CircleCI configuration
  - Adds Coveralls coverage reporting
  - Adds ESLint, Prettier, Jest
  - Adds License, Code of Conduct, Contributing
  - Adds Slack integrations
  - Adds custom logger using Winston

## Pull Request Checklist:

  - [x] My submission passes all tests.
  - [x] I have lint my code locally prior to submission.
  - [x] I have written new tests for my core changes (where applicable).
  - [x] I have referenced all useful information (issues, tasks, etc) in the pull request.

### CODE REVIEW

File: bot/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+require("dotenv").config();
+
+const logger = require("./utils/logger");
+const express = require("express");
+const path = require("path");
```
Comment (joeytepp) [inline bot/index.js]: Do we need this here? Doesn't look like it's used

Comment (RobertWSaunders) [reply bot/index.js]: Nope, I'll remove it, good ol' copy and paste, nice catch! 👍

Comment (RobertWSaunders) [reply bot/index.js]: Updated!

File: package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    "type": "git",
+    "url": "https://github.com/qhacks/qhacks-bot.git"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "pretty-quick --staged"
+    }
+  },
+  "scripts": {
+    "dev": "NODE_ENV=development nodemon bot/index.js",
```
Comment (joeytepp) [inline package.json]: Nitpick but could we change `bot/index.js` to just `bot`? Feel like it would look nicer

Comment (RobertWSaunders) [reply package.json]: Updated!

**Comment by joeytepp** [comment]
Looking really good so fired up to work on this :fire::fire: A few comments about a couple small things, overall gonna be a huge addition to the qhacks codebase!

**Comment by RobertWSaunders** [comment]
@joeytepp Should all be updated!

**Pull request merged**